src/main/kotlin/no/nav/syfo/service/BehandlendeEnhetService.kt

```diff
@@ -0,0 +1,84 @@
+package no.nav.syfo.service
+
+import net.logstash.logback.argument.StructuredArguments
+import no.nav.syfo.client.ArbeidsFordelingClient
+import no.nav.syfo.client.ArbeidsfordelingRequest
+import no.nav.syfo.log
+import no.nav.syfo.pdl.model.PdlPerson
+import no.nav.syfo.util.LoggingMeta
+import no.nav.tjeneste.pip.egen.ansatt.v1.EgenAnsattV1
+import no.nav.tjeneste.pip.egen.ansatt.v1.WSHentErEgenAnsattEllerIFamilieMedEgenAnsattRequest
+import no.nav.tjeneste.virksomhet.person.v3.binding.HentGeografiskTilknytningPersonIkkeFunnet
+import no.nav.tjeneste.virksomhet.person.v3.binding.HentGeografiskTilknytningSikkerhetsbegrensing
+import no.nav.tjeneste.virksomhet.person.v3.binding.PersonV3
+import no.nav.tjeneste.virksomhet.person.v3.informasjon.NorskIdent
+import no.nav.tjeneste.virksomhet.person.v3.informasjon.PersonIdent
+import no.nav.tjeneste.virksomhet.person.v3.informasjon.Personidenter
+import no.nav.tjeneste.virksomhet.person.v3.meldinger.HentGeografiskTilknytningRequest
+
+class BehandlendeEnhetService(val personV3: PersonV3, val egenAnsattV1: EgenAnsattV1, val arbeidsfordelingClient: ArbeidsFordelingClient) {
+
+    suspend fun getBehanldendeEnhet(pasient: PdlPerson, loggingMeta: LoggingMeta): String? {
+        val geografiskTilknytning = getGeografiskTilknytning(pasient, loggingMeta)
+        val egenAnsatt = isEgenAnsatt(pasient)
+        return getBehandlendeId(pasient, geografiskTilknytning, egenAnsatt, loggingMeta)
+    }
+
+    private suspend fun getBehandlendeId(pasient: PdlPerson, geografiskTilknytning: String?, egenAnsatt: Boolean, loggingMeta: LoggingMeta): String {
+        val result = arbeidsfordelingClient.finnBehandlendeEnhet(getArbeidsfordelingRequest(pasient, geografiskTilknytning, egenAnsatt))
+        if (result?.firstOrNull()?.enhetId == null) {
+            log.warn("arbeidsfordeling fant ingen nav-enheter {}", StructuredArguments.fields(loggingMeta))
+        }
+        return result?.firstOrNull()?.enhetNr
+                ?: "0393"
+    }
+
+    private fun isEgenAnsatt(pasient: PdlPerson) =
```

Comment:
Vi får ikke dette fra PDL heller..?

Reply:
Nei. "Egen ansatt er ansett som HR informasjon. Det jobbes med å etablere "Egen ansatt" som en del av ny Orgmaster. Frem til Orgmaster er på plass er TPS fortsatt master."

Reply:
Kjipt altså :( Trodde de hadde innført et nytt konsept, "skjermet", for egne ansatte, men mulig jeg husker feil da :(